internal/service/privatelinkendpointservicedatafederationonlinearchive/resource_privatelink_endpoint_service_data_federation_online_archive_test.go

```diff
@@ -92,13 +168,33 @@ func checkExists(resourceName string) resource.TestCheckFunc {
 	}
 }
 
-func resourceConfigBasic(projectID, endpointID string) string {
+func resourceConfigBasic(projectID, endpointID, comment string) string {
 	return fmt.Sprintf(`
 	resource "mongodbatlas_privatelink_endpoint_service_data_federation_online_archive" "test" {
 	  project_id				= %[1]q
 	  endpoint_id				= %[2]q
 	  provider_name				= "AWS"
-	  comment					= "Terraform Acceptance Test"
+	  comment					= %[3]q
+	}
+	`, projectID, endpointID, comment)
+}
+
+func toCustomerEndpointDNSName(endpointID string) string {
+	// Found in `.AWS.dev.us-east-1`:  https://github.com/10gen/mms/blob/85ec3df92711014b17643c05a61f5c580786556c/server/conf/data-lake-endpoint-services.json
+	serviceName := "vpce-svc-0a7247db33497082e"
+	return fmt.Sprintf("%s-8charsra.%s.us-east-1.vpce.amazonaws.com", endpointID, serviceName)
+}
```

Comment:
This should be a random alphanumeric 8 character string created by aws.
Therefore, I created the `8charsra` (`char=characters` and `ra=random` ) 😅. I'll add a comment

Reply:
just to be sure I understand, if its created by aws would this have to be modified each time it is run? If that is the case I would favor extracting to an env variable to it is clear what needs to be adjusted (or provided externally). Otherwise happy to leave as is.

Reply:
You are right. Although the test was passing with any random value (I don't think the API checks that it matches the vpc endpoint dns name), this might not work in practice.
Using an env variable also enables us to run this test in different AWS region/environments where serviceName is different.

Changed in 09c4b80